src/Sarif/RuleUtilities.cs

```diff
@@ -24,33 +22,22 @@ public static Result BuildResult(ResultLevel level, IAnalysisContext context, Re
                 throw new ArgumentNullException(nameof(arguments));
             }
 
-            string[] messageArguments = arguments;
```

Comment:
No need for the temp, since we're not going to modify `arguments`. Just use `arguments` directly.